### DISCUSSION TITLE
Finally task pipelinerrun status

### DISCUSSION THREAD

**Discussion by gdisanto** [tektoncd/pipeline#3913] [Q&A] [answered]
Hi,
Is that possible to check the status of the pipelinerun in a task under the finally branch?
I know that I can check the status of the individual tasks in the pipelinerun and I could potentially check the status of the last one to determine if the pipelinerun was successful or not. 
My issue is that I have a pipeline whose last two tasks can be executed or not depending on some conditions. I would need to put another finally task to check if the previous one failed as well and it the finally part gets really messy. 
It would be nice to be able to check the overall status of the pipelinerun to just notify if it was successful or it failed.

**Comment by jerop** [comment, marked as answer]
hi @gdisanto

@pritidesai has been working on this issue:

- issue: https://github.com/tektoncd/pipeline/issues/3806
- pr: https://github.com/tektoncd/pipeline/pull/3817
- tep: https://github.com/tektoncd/community/blob/main/teps/0049-aggregate-status-of-dag-tasks.md

it's in the [milestone for this month's release](https://github.com/tektoncd/pipeline/milestone/43)

**Reply by gdisanto** [reply]
Hi @jerop and @pritidesai,
This is exactly what I was looking for! Thanks a lot, hopefully I can try it soon :).
